sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -582,6 +582,109 @@ case class StringLocate(substr: Expression, str: Expression, start: Expression)
     }
   }
 
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    if (start.foldable) {
```

Comment:
I think we do not need these special cases for StringLocate, because you do not need any prepare to call UTF8String.indexOf(substr, start)